lambdas/s3hash/src/t4_lambda_s3hash/__init__.py

```diff
@@ -306,12 +311,15 @@ async def compute_checksum_legacy(location: S3ObjectSource) -> Checksum:
         async for chunk in stream.content.iter_any():
             hashobj.update(chunk)
 
-    return Checksum.singlepart(hashobj.digest())
+    return Checksum.legacy(hashobj.digest())
 
 
 async def compute_checksum(location: S3ObjectSource) -> ChecksumResult:
     obj_attrs = await get_obj_attributes(location)
     if obj_attrs:
+        if obj_attrs["ObjectSize"] == 0:
+            return ChecksumResult(checksum=Checksum.empty())
+
```

Comment:
I think you don't need this
you could just unindent the check below

Reply:
i don't quite get what you mean

Reply:
you could check object size with a single condition regardless whether you have it from get_object_attributes() or head_object()

Reply:
sure, but i need to do that before calling `get_compliant_checksum`, bc that may return a wrong checksum (double-hashed)

Reply:
well, then it might make sense to have that check in `get_compliant_checksum()` or add assert 
if you go we assert you can do here something like

1. object_size, etag = ...
2. if object_size == 0: ...
3. if obj_attrs is not None: ...